### PR TITLE
Add conversion for SDFormat Links

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

## Summary
Builds on top of #2 to add conversion for links.

## Test it
Run unit tests

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

File: sdformat_mjcf/sdformat_mjcf/converters/link.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    for ci in range(link.collision_count()):
+        col = link.collision_by_index(ci)
+        if col.geometry() is not None:
+            add_collision(body, col, pose_resolver=pose_resolver)
+
+    for vi in range(link.visual_count()):
+        col = link.visual_by_index(vi)
+        if col.geometry() is not None:
+            add_visual(body, col, pose_resolver=pose_resolver)
+
```
Comment (ahcorde) [inline sdformat_mjcf/sdformat_mjcf/converters/link.py:86]:
```suggestion
    for vi in range(link.visual_count()):
        vis = link.visual_by_index(vi)
        if vis.geometry() is not None:
            add_visual(body, vis, pose_resolver=pose_resolver)

```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: sdformat_mjcf/sdformat_mjcf/converters/link.py (hunk 1 above)
Comment (azeey) [reply sdformat_mjcf/sdformat_mjcf/converters/link.py]: https://github.com/gazebosim/gz-mujoco/pull/5/commits/2e54b846be937d328dff7d04ac8da1edf7dfc0fa

**Comment by azeey** [comment]
I've added more tests in https://github.com/gazebosim/gz-mujoco/pull/5/commits/3f481f26708e0faf59fbee4d33fd593b5a417641 to increase test coverage. @ahcorde PTAL.

**Review by ahcorde — APPROVED** [review]

**Review by scpeters — COMMENTED** [review]
this looks reasonable to me. I would just consider adding an expectation on the exact value of the `fullinertia` attribute (see https://github.com/gazebosim/gz-mujoco/commit/175b5d19b12370753edfb2348a912aa03adbbc78)

**Comment by azeey** [comment]
> this looks reasonable to me. I would just consider adding an expectation on the exact value of the `fullinertia` attribute (see [175b5d1](https://github.com/gazebosim/gz-mujoco/commit/175b5d19b12370753edfb2348a912aa03adbbc78))

Thanks Steve! Are those values based on a known shape?

**Comment by scpeters** [comment]
> Thanks Steve! Are those values based on a known shape?

Yes, they are based on a box with relative dimensions 1x4x9. I started with the values in [this test case](https://github.com/gazebosim/gz-math/blob/ign-math6/src/MassMatrix3_TEST.cc#L763-L771) but increased the mass by a factor of 32 (12 -> 384) to avoid decimals in the inertial values (aside from `sqrt(6)`). Looking again the values could all be reduced by a factor of 4, but I think it works either way.

**Pull request merged**